lang/python/python-pytest/Makefile

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (C) 2020 CZ.NIC, z. s. p. o. (https://www.nic.cz/)
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=python-pytest
+PKG_VERSION:=5.4.1
+PKG_RELEASE:=1
+
+PYPI_NAME:=pytest
+PKG_HASH:=84dde37075b8805f3d1f392cc47e38a0e59518fb46a431cfdaf7cf1ce805f970
+
+PKG_MAINTAINER:=Jan Pavlinec <jan.pavlinec@nic.cz>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+
+include ../pypi.mk
+include $(INCLUDE_DIR)/package.mk
+include ../python3-package.mk
+
+define Package/python3-pytest
+  SUBMENU:=Python
+  SECTION:=lang
+  CATEGORY:=Languages
+  TITLE:=Python testing framework
+  URL:=https://docs.pytest.org/en/latest/
+  DEPENDS:= \
+	+python3-light \
+	+python3-atomicwrites \
+	+python3-more-itertools \
+	+python3-py \
+	+python3-attrs \
+	+python3-pluggy \
+	+python3-packaging \
+	+python3-wcwidth
+  VARIANT:=python3
```

Comment:
Is "variant" still needed after the python2 removal was done?

Reply:
There is still a dependency on `VARIANT:=python3` in python3-package.mk.

@commodo How do you feel about keeping/removing the variant mechanism? I can see arguments for either option.

(Apologies for hijacking this comment thread for a Python discussion :joy:)

Reply:
fine to remove `VARIANT` [from my side];
but we should make it aware to people that it is required in 19.x and 18.x releases;
no good idea about how to do it: document it? add a comment somewhere visible?

i've seen that sometimes people want some newer Python packages [or packages that are only in master] in releases and they ask for them, or try to backport them directly and get stuck, and open issues on Github;

either we leave the `VARIANT` mechanism for one last release [20.X] or we remove it now and we keep commenting to them that they should put the VARIANT variable [when backporting];
either is fine from my side;

that's pretty much the only things [at this point in time] about `VARIANT` that i see that is worth worrying about

Reply:
The backporting aspect is a factor suggesting that the VARIANT is left in place for now.

In master there might maybe be a default statement/value in python3-package.mk, so that the variant would be unnecessary for new packages (which will not be backported to 19.07).  That would enable a gradual removal of VARIANT from master.

Reply:
I'm not too concerned about backporting; we can add a note in the readme. Having a section that documents major changes between releases is probably a good idea anyway.

I'm not sure it's possible/easy to have a default value for `VARIANT`... I know there is a `DEFAULT_VARIANT` but I don't believe it affects package compilation (actually I'm not really sure what effect it has, if any).

I'm thinking of refactoring python3-package.mk so that different packages can use different aspects of its functionality (calling setup.py, bytecode compilation, shebang manging, etc.). I'll probably take out the variant mechanism as part of that and also do a treewide update.

Until this refactoring is done (whenever it may be :joy:) I think Python package PRs should continue including `VARIANT:=python3`. Would this be ok?

Reply:
> Until this refactoring is done (whenever it may be joy) I think Python package PRs should continue including `VARIANT:=python3`. Would this be ok?

fine by me